### DISCUSSION TITLE
Q3 of Day 3

### DISCUSSION THREAD

**Discussion by heejeongch** [I-JUNNYLAND-I/algorithm#33] [Q&A] [answered]
1. 우선 주니님이 힌트를 준거에 대해서 어떻게 다루어야할지 감이 안잡힙니다.
2. 스택으로 풀려고 했으나 반복문에서 어떻게 계속 팝을 시켜주어야할지 감이 안잡힙니당.

**Comment by Junnyjun** [comment, marked as answer]
스택이기 보다

 [ ] ( ) { } 일 경우
 []상쇄 ()상쇄 {}상쇄 하면 됩니다.
 
 ] 가 발견되었다면 stack을 꺼내 [ 가 있는지 보면되지 않을까요?

**Reply by Junnyjun** [reply]
```java

if ( s == '(' ){
 stack.push(s)
}else if( s == ')' && stack.peek() == '(' ){
 st.pop()
}
```
